services/dashboard/views/apps/streamingapp/dag.js

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed under the Apache License, Version 2.0
+ * See accompanying LICENSE file.
+ */
+
+angular.module('dashboard')
+
+  .config(['$stateProvider',
+    function($stateProvider) {
+      'use strict';
+
+      $stateProvider
+        .state('streamingapp.dag', {
+          url: '/dag',
+          templateUrl: 'views/apps/streamingapp/dag.html',
+          controller: 'StreamingAppDagCtrl'
+        });
+    }])
+
+  .controller('StreamingAppDagCtrl', ['$scope', '$state', '$modal', '$contextmenu', '$visNetworkStyle',
```

Comment:
Controller should be marked for scalajs - in general where ever common (REST) types are processed, it should be in scalajs for type checking.